### PR TITLE
update GA

### PR DESCRIPTION
**Pull request by ermiaSparkle**
closes https://github.com/sparkletown/internal-sparkle-issues/issues/928

### CODE REVIEW

File: src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx [hunk 1]
```diff
@@ -105,9 +105,11 @@ export const RoomModalContent: React.FC<RoomModalContentProps> = ({
...
     if (venue.template === VenueTemplate.partymap && isExternalUrl(room.url)) {
       logEventGoogleAnalytics({
-        eventCategory: "PARTMAP_WITH_EXTERNAL_LINK",
-        eventAction: "ENTER_PARTMAP_" + venueName.toUpperCase(),
-        eventLabel: room.title,
+        eventName: "PARTMAP_WITH_EXTERNAL_LINK",
+        eventAction: {
+          VenueId: venueName,
+          Link: room.url,
+        },
```
Comment (mike-lvov) [inline src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx:112]: venueName,
roomUrl
userId

Should be the props

**Review by mike-lvov — CHANGES_REQUESTED** [review]

File: src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx (hunk 1 above)
Comment (ermiaSparkle) [reply src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx:112]: Why do we need userId?

Comment (ermiaSparkle) [reply src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx]: fixed in 470765927067ef5fd74350608824763fd7075001

File: src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx [hunk 2]
```diff
@@ -79,6 +80,8 @@ export const RoomModalContent: React.FC<RoomModalContentProps> = ({
 }) => {
   const { name: venueName, showSchedule = DEFAULT_SHOW_SCHEDULE } = venue;
 
+  const { user } = useUser();
```
Comment (mike-lvov) [inline src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx:83]:
```suggestion
  const { userId } = useUser();
```

File: src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx [hunk 3]
```diff
@@ -105,12 +108,15 @@ export const RoomModalContent: React.FC<RoomModalContentProps> = ({
...
       logEventGoogleAnalytics({
-        eventCategory: "PARTMAP_WITH_EXTERNAL_LINK",
-        eventAction: "ENTER_PARTMAP_" + venueName.toUpperCase(),
-        eventLabel: room.title,
+        eventName: "PARTMAP_WITH_EXTERNAL_LINK",
+        eventAction: {
+          VenueId: venueName,
+          RoomUrl: room.url,
+          UserId: user?.uid,
+        },
```
Comment (mike-lvov) [inline src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx:116]:
```suggestion
        eventName: "PARTMAP_WITH_EXTERNAL_LINK",
        eventAction: {
          venueName,
          userId,
          roomUrl: room.url,
        },
```

File: src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx [hunk 4]
```diff
@@ -105,12 +108,15 @@ export const RoomModalContent: React.FC<RoomModalContentProps> = ({
...
       });
     }
-  }, [_enterRoomWithSound, venueName, room, venue]);
+  }, [_enterRoomWithSound, venueName, room, venue, user]);
```
Comment (mike-lvov) [inline src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx:119]:
```suggestion
  }, [_enterRoomWithSound, venueName, room, venue, userId]);
```

File: src/index.tsx [hunk 5]
```diff
@@ -79,6 +78,7 @@ if (LOGROCKET_APP_ID) {
 
 const firebaseApp = firebase.initializeApp(FIREBASE_CONFIG);
 firebaseApp.analytics();
+
```
Comment (mike-lvov) [inline src/index.tsx:81]:
```suggestion
```

File: src/utils/googleAnalytics.ts [hunk 6]
```diff
@@ -1,29 +1,19 @@
...
-    pageview(window.location.pathname + window.location.search);
-  }
-};
 // @debt properly type the props, not just string
 type GAEvent = {
-  eventCategory: string;
-  eventAction: string;
-  eventLabel: string;
+  eventName: string;
+  // eventCategory: string;
```
Comment (mike-lvov) [inline src/utils/googleAnalytics.ts:8]:
```suggestion
```

**Review by mike-lvov — CHANGES_REQUESTED** [review]

**Review by mike-lvov — APPROVED** [review]
LGTM

**Pull request merged**